cargo-dist/src/tasks.rs

```diff
@@ -736,6 +741,16 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
             DirtyMode::AllowList(allow_dirty.clone().unwrap_or(vec![]))
         };
 
+        let system_dependencies =
+            workspace_metadata
+                .system_dependencies
```

Comment:
hmm wait are the package-local dependencies being ignored? it makes sense that we'd want to gather a global final list to emit in the CI, but I would think it would want to be "union together all dependencies for each package we've decided we're building right now")

Reply:
Hmmm, right. Good point - I hadn't really handled the interaction between the two, but I should.

Reply:
Updated to do this, though it looks like we don't have integration tests that make use of package-local stuff so I didn't add an integration test.